pkg/apiserver/apiserver.go

```diff
@@ -210,13 +210,13 @@ func logStackOnRecover(s runtime.NegotiatedSerializer, panicReason interface{},
 	errorNegotiated(apierrors.NewGenericServerResponse(http.StatusInternalServerError, "", api.Resource(""), "", "", 0, false), s, unversioned.GroupVersion{}, w, &http.Request{Header: headers})
 }
 
-func InstallServiceErrorHandler(s runtime.NegotiatedSerializer, container *restful.Container, requestResolver *RequestInfoResolver, apiVersions []string) {
+func InstallServiceErrorHandler(s runtime.NegotiatedSerializer, container *restful.Container) {
 	container.ServiceErrorHandler(func(serviceErr restful.ServiceError, request *restful.Request, response *restful.Response) {
-		serviceErrorHandler(s, requestResolver, apiVersions, serviceErr, request, response)
+		serviceErrorHandler(s, serviceErr, request, response)
 	})
 }
 
-func serviceErrorHandler(s runtime.NegotiatedSerializer, requestResolver *RequestInfoResolver, apiVersions []string, serviceErr restful.ServiceError, request *restful.Request, response *restful.Response) {
+func serviceErrorHandler(s runtime.NegotiatedSerializer, serviceErr restful.ServiceError, request *restful.Request, response *restful.Response) {
```

Comment:
For the moment this was syntactical pruning only. I would like to get rid of the service handler installation from the api group installer (https://github.com/kubernetes/kubernetes/pull/33368/files#diff-21175d1496189457be1a811db6ce0458L398).

Following your idea, we could create "group/version" strings of all installed api groups after they have been installed and use those (i.e. in the beginning of `Run()`). The current list of versions created after each group installation and then overwritten, does not make sense.

Which nicer message do you have in mind?